src/rrules/builtins.jl

```diff
@@ -219,6 +219,9 @@ function rrule!!(f::CoDual{typeof(bitcast)}, t::CoDual{Type{T}}, x) where {T}
     v = bitcast(T, _x)
     if T <: Ptr && _x isa Ptr
         dv = bitcast(Ptr{tangent_type(eltype(T))}, tangent(x))
+    elseif T <: Ptr && _x isa Union{Int,UInt}
+        @warn "Creating Core.bitcast's primal tangent from a raw integer to a pointer is potentially unsafe."
+        dv = bitcast(Ptr{tangent_type(eltype(T))}, _x)
```

Comment:
```suggestion
        throw(error("Creating `Core.bitcast`'s primal tangent from a raw integer is unsafe."))
```

Reply:
> but with this PR we can handle AD for programs where a valid address given as an Int/UInt form is being converted to a Ptr to use.

I don't think there's a use case for this. I suggest we forbid this with an informative error message. 

We can revisit this if there is a motivating example.